doc/linters.md

```diff
@@ -1366,6 +1367,18 @@ Mismatched bracket: found an opening [ and a closing ) on line 1
 Mismatched bracket: found an opening [ on line 1 and a closing )
 ```
 
+### Threaded function form
+
+*Keyword:* `:threaded-fn-form`.
+
+*Description:* warn when directly supplying a function literal as one of the `forms` (n.b. not the first "expression" argument) to the `->` "thread-first" or `some->` macro. Although this can compile (when a symbol literal is threaded), threading into the `name` slot of an anonymous function is almost always not intended. "Double wrapping" is usually what is intended (e.g. `(-> foo ((fn [x] ...)))`).
+
+*Default level:* `warning`.
```

Comment:
New linters are usually disabled by default, to incubate while people test it out and not break existing setups for people.